kernel/data-pipeline/dialect/opengauss/src/main/java/org/apache/shardingsphere/data/pipeline/opengauss/ingest/wal/decode/MppdbDecodingPlugin.java

```diff
@@ -78,15 +76,28 @@ public AbstractWALEvent decode(final ByteBuffer data, final BaseLogSequenceNumbe
     
     private AbstractWALEvent decodeDataWithTX(final String dataText) {
         AbstractWALEvent result = new PlaceholderEvent();
-        if (dataText.startsWith("BEGIN")) {
-            int beginIndex = dataText.indexOf("BEGIN") + "BEGIN".length() + 1;
-            result = new BeginTXEvent(Long.parseLong(dataText.substring(beginIndex)));
-        } else if (dataText.startsWith("COMMIT")) {
-            int commitBeginIndex = dataText.indexOf("COMMIT") + "COMMIT".length() + 1;
-            int csnBeginIndex = dataText.indexOf("CSN") + "CSN".length() + 1;
-            result = new CommitTXEvent(Long.parseLong(dataText.substring(commitBeginIndex, dataText.indexOf(' ', commitBeginIndex))), Long.parseLong(dataText.substring(csnBeginIndex)));
-        } else if (dataText.startsWith("{")) {
-            result = readTableEvent(dataText);
+        if (majorVersion <= 2) {
+            if (dataText.startsWith("BEGIN")) {
+                int beginIndex = dataText.indexOf("BEGIN") + "BEGIN".length() + 1;
+                result = new BeginTXEvent(Long.parseLong(dataText.substring(beginIndex)), null);
+            } else if (dataText.startsWith("COMMIT")) {
+                int commitBeginIndex = dataText.indexOf("COMMIT") + "COMMIT".length() + 1;
+                int csnBeginIndex = dataText.indexOf("CSN") + "CSN".length() + 1;
+                result = new CommitTXEvent(Long.parseLong(dataText.substring(commitBeginIndex, dataText.indexOf(' ', commitBeginIndex))), Long.parseLong(dataText.substring(csnBeginIndex)));
+            } else if (dataText.startsWith("{")) {
+                result = readTableEvent(dataText);
+            }
+        } else {
+            if (dataText.startsWith("BEGIN")) {
+                int beginIndex = dataText.indexOf("CSN:") + "CSN:".length() + 1;
+                int endIndex = dataText.indexOf("first_lsn") - 1;
+                result = new BeginTXEvent(null, Long.parseLong(dataText.substring(beginIndex, endIndex)));
+            } else if (dataText.startsWith("commit")) {
+                int beginIndex = dataText.indexOf("xid:") + "xid:".length() + 1;
+                result = new CommitTXEvent(Long.parseLong(dataText.substring(beginIndex)), null);
+            } else if (dataText.startsWith("{")) {
+                result = readTableEvent(dataText);
+            }
```

Comment:
1, Could we extract 2 methods, one is for openGauss 2, and another is for openGauss 3+

2, `majorVersion <= 2` could be `majorVersion < 3`, use the same expression as other places